File: sam/fizzbuzz.py
print "\n".join(map(lambda y: y[1].lstrip('0123456789') + "buzz" if not y[0] % 5 else y[1] + '', [(x,y) for x, y in enumerate(map(lambda x: x[1] + "fizz" if not x[0] % 3 else x[1] + str(x[0]), [(x,y) for x, y in enumerate([''] * 101)]))])[1:])

File: sam/fizzbuzz.hs
import System.IO
import Data.List

fizz :: [Int] -> [(Int, String)]
fizz l = 
  zip l [if x `mod` 3 == 0 then "fizz" else "" | x <- l]
    
buzz :: (Int, String) -> String
buzz t = 
  let s = snd t
  in if (fst t) `mod` 5 == 0
     then s ++ "buzz"
     else if null s
          then s ++ (show (fst t))
          else s ++ ""
            
fizzbuzz :: [Int] -> [String]
fizzbuzz l = 
  map buzz (fizz l)

main = 
  let s = fizzbuzz [1..100]
  in putStrLn (intercalate "\n" s)